PhoneBook.Domain/Location.cs

```diff
@@ -53,30 +52,30 @@ protected Location (string street, string number, string city, Country country,
 
     public static Location GetObject(ObjectID objid)
     {
-      return DomainObject.GetObject<Location>(objid);
+      return GetObject<Location> (objid);
     }
 
-    public static Location[] GetLocations ()
+    public static IEnumerable<Location> GetLocations ()
```

Comment:
Since the result is already evaluated and there is no reference kept to the array, returning the Array instead of a sequence is beneficial:
* The user knows that the collection type can be iterated multiple times without reexecuting the query, plus the user can use Length, etc.
* Other APIs exist that don't accept IEnumerable, e.g. Boclist Value collection is typed to IList. 

Personally, I can see the appeal of IEnumerable, but only where the intention of the API is to just iterate the result anyway, or the values are being produced on demand anyway.